src/main/java/org/zalando/nakadi/repository/db/EventTypeDbRepository.java

```diff
@@ -48,10 +48,15 @@ public void saveEventType(final EventType eventType) throws InternalNakadiExcept
 
     @Override
     public EventType findByName(final String name) throws NoSuchEventTypeException {
-        final String sql = "SELECT et_topic, et_event_type_object FROM zn_data.event_type WHERE et_name = ?";
+        final String sql = "SELECT et_topic, et_event_type_object, et_deleted " +
+                "FROM zn_data.event_type WHERE et_name = ?";
 
         try {
-            return jdbcTemplate.queryForObject(sql, new Object[]{name}, new EventTypeMapper());
+            final EventType eventType = jdbcTemplate.queryForObject(sql, new Object[]{name}, new EventTypeMapper());
+            if (eventType.isDeleted()) {
```

Comment:
I would say that this could be done using plain query. Maybe adding another condition `et_deleted = false`? Then it would return an empty set.